scps/SCP-002.md

```diff
@@ -0,0 +1,69 @@
+## SCP-002: Datastore P2P interface
+
+This proposal is a system that allows stores to create custom p2p channels to other peers interested in a common orbitdb datastore.
+
+**Definitions**:
+* Asking peer: The peer initiating the connection
+* Responding peer: The peer responding to the initiation.
+* The network: Cloud of peers all interconnected interested in the datastore in question
+* Broadcast channel: Global pubsub channel of the corresponding datastore.
+* Application specific P2P API: Interface diagrammed by higher level code implementing `orbit-db-store`; Defined by the store type.
+
+### Hypothetical model N1
+> Note all p2p traffic is over IPFS pubsub
+
+**Flow**:
+
+1. Asking peer joins store broadcast channel. (Example: Asking peer is just starting up and gets connected to the network)
+2. Responding peer broadcasts a small message to all the other peers containing information regarding what diagrams/protocol it supports. Including Public key for encryption, if needed, and a forwarding pubsub channel ID (a channel where asking peers can send messages directly to the responding peer, derived from the peerId). The forwarding channel isn't designed to be a back and forth between a responding peer and an asking peer.  Only a temporary channel for negotiating a direct peer-to-peer connection.
+3. Asking peer sends an establishment request to the forwarding channel, containing the necessary information to determine a direct p2p channel ID.
+4. In the newly created pubsub channel, protocol handshakes are done, encryption is established, exchange of supported p2p application specific API calls, Secondary application specific handshakes completed. The rest of the process is diagrammed by the application specific API.
+
+**Advantages**
+* Relies on a technology already used in orbitdb.
+* Pubsub relays can help with routing packets to peers behind a firewall.
+
+**Disadvantages**
+* No P2P transport security out of the box. Higher chance of an adversory capturing P2P messages
```

Comment:
At least we have handshake security with the Noise protocol. Wondering what else we can utilize. However, with the externally-pluggable requirement we should probably handle this at the `orbit-db-pubsub` level

Reply:
I am unaware the Noise protocol can be used over pubsub? Are you talking about noise protocol in regards to libp2p? Could you provide some more clarification?